tutorial/quickstart.md

```diff
@@ -0,0 +1,459 @@
+
+These are the contents of the current directory:
+```bash
+$ tree
+.
+├── mod.cpython-36m-x86_64-linux-gnu.so
+├── mod.py
+└── __pyccel__
+    ├── build
+    │   ├── lib.linux-x86_64-3.6
+    │   └── temp.linux-x86_64-3.6
+    │       └── mod_wrapper.o
+    ├── mod.c
+    ├── mod.h
+    ├── mod.o
+    ├── mod_wrapper.c
+    └── setup_mod.py
+
+4 directories, 10 files
+
+```
+
+This is the C file `mod.c` generated by Pyccel:
+```c
+#include <mod.h>
+#include <math.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+/*........................................*/
+int64_t factorial(int64_t n)
+{
+    int64_t Out_0001;
+    if (n <= 0)
+    {
+        Out_0001 = 1;
+        return Out_0001;
+    }
+    else if (1)
+    {
+        Out_0001 = n * factorial(n - 1);
+        return Out_0001;
+    }
+}
+/*........................................*/
+
+/*........................................*/
+int64_t binomial_coefficient(int64_t n, int64_t k)
+{
+    int64_t num;
+    int64_t den;
+    int64_t Out_0001;
+    num = factorial(n);
+    den = factorial(k) * factorial(n - k);
+    Out_0001 = floor((double)(num) / (double)(den));
+    return Out_0001;
+}
+/*........................................*/
+```
+And this is the header file `mod.h`:
+```c
+#ifndef MOD_H
+#define MOD_H
+
+#include <stdlib.h>
+#include <stdint.h>
+
+int64_t factorial(int64_t n);
+
+int64_t binomial_coefficient(int64_t n, int64_t k);
+
+#endif // MOD_Hy
+```
+
+#### Example 3: matrix-matrix multiplication with OpenMP
+
+Let's now see a more complicated example, where the Python module `mod.py` contains a function that performs the matrix-matrix multiplication between two arrays `a` and `b`, and writes the result into the array `c`:
+
+-   The three function's arguments are 2D Numpy arrays of double-precision floating point numbers
+-   Matrices `a` and `c` use C ordering (row-major), matrix `b` uses Fortran ordering (column-major)
+-   Since matrix `c` is modified by the function, it has `intent(inout)` in Fortran
+-   Comments starting with `#$ omp` are translated to OpenMP pragmas
+
+```python
+def matmul(a: 'float[:,:](order=C)',
+           b: 'float[:,:](order=F)',
+           c: 'float[:,:](order=C)'):
+
+    m, p = a.shape
+    q, n = b.shape
+    r, s = c.shape
+
+    if p != q or m != r or n != s:
+        return -1
+
+#$ omp parallel
+#$ omp for schedule(runtime)
+    for i in range(m):
+        for j in range(n):
+            c[i, j] = 0.0
+            for k in range(p):
+                c[i, j] += a[i, k] * b[k, j]
+#$ omp end parallel
+
+    return 0
+```
+We now translate this file to Fortran, and compile it to a Python C extension module, using the command
+```bash
+$ pyccel mod.py --language fortran
+```
+The flag `--language fortran` may be omitted, as Pyccel uses Fortran as the default backend language.
```

Comment:
Is it worth mentioning here that arrays are not yet fully supported in C?

Reply:
I am not sure that we could claim that they are fully supported in Fortran either! :sweat_smile:

In any case our array support in C is getting very close to Fortran, I don't think we need to be too specific here. The more we write and the more we have to maintain!

By the way, I've run the quicksort example with C, but I got a runtime error (segmentation fault).

Reply:
I have found the problem. It is linked to the `FloorDiv` operator. I am opening 2 issues about it. The example works in c if you use:
`int((lo+hi)/2)` instead of `(lo+hi)//2`

Reply:
Thanks for looking into this! I will keep the example as it is, as I am sure that we're going to fix the problem soon. Maybe this code should be added to our unit tests.